tsinfer/formats.py

```diff
@@ -1987,6 +1988,7 @@ def add_ancestor(self, start, end, time, focal_sites, haplotype):
             focal_sites=focal_sites,
             haplotype=haplotype,
         )
+        return self.item_writer.total_items - 1
```

Comment:
Would be better for the item_writer to return the index, I think? Keeps encapsulation better, plus resolves any threading issues since you'd expect the writer to hold any locks needed during add.

Reply:
Looks like it [already returns the index](https://github.com/tskit-dev/tsinfer/blob/8877a8f76af1bf5329bcd3a9bcb36ab2d7221e8b/tsinfer/formats.py#L214), so it's even more trivial than I thought!